File: src/main/java/com/letgo/book/infrastructure/persistence/mapping/BookTitleConverter.java
package com.letgo.book.infrastructure.persistence.mapping;

import com.letgo.book.domain.BookTitle;

import javax.persistence.AttributeConverter;
import javax.persistence.Converter;

@Converter
final public class BookTitleConverter implements AttributeConverter<BookTitle, String> {
    @Override
    public String convertToDatabaseColumn(BookTitle attribute) {
        return attribute == null ? null : attribute.value();
    }

    @Override
    public BookTitle convertToEntityAttribute(String dbData) {
        return dbData == null ? null : BookTitle.create(dbData);
    }
}

File: src/main/java/com/letgo/book/infrastructure/persistence/mapping/BookTitleConverter.kt
package com.letgo.book.infrastructure.persistence.mapping

import com.letgo.book.domain.BookTitle
import javax.persistence.AttributeConverter
import javax.persistence.Converter

@Converter
class BookTitleConverter : AttributeConverter<BookTitle, String> {
    override fun convertToDatabaseColumn(attribute: BookTitle): String {
        return attribute.value()
    }

    override fun convertToEntityAttribute(dbData: String?): BookTitle? {
        return if (dbData == null) null else BookTitle.create(dbData)
    }
}
